Parse/src/main/java/com/parse/ParseObject.java

```diff
@@ -1433,24 +1447,22 @@ private ParseRESTObjectCommand currentSaveEventuallyCommand(
         // Merge the data from the failed save into the next save.
         ParseOperationSet nextOperation = opIterator.next();
         nextOperation.mergeFrom(operationsBeforeSave);
+        if (store != null) {
+          task = task.continueWithTask(new Continuation<Void, Task<Void>>() {
+            @Override
+            public Task<Void> then(Task<Void> task) throws Exception {
+              if (task.isFaulted()) {
+                return Task.forResult(null);
```

Comment:
This is to not propagate the exception in case fetching the store [here](https://github.com/parse-community/Parse-SDK-Android/pull/831/files#diff-8e759e84d8f2c4f7fa6bc715e3786808R1432) fails. If exception was propagated, existing ParseInstallationTests would fail ([testMissingRequiredFieldWhenSaveAsync](https://github.com/DanGTZ/Parse-SDK-Android/blob/70516c3fb3452c521e4ce26e9287d806bff0fda6/Parse/src/test/java/com/parse/ParseInstallationTest.java#L125) and [testObjectNotFoundWhenSaveAsync](https://github.com/DanGTZ/Parse-SDK-Android/blob/70516c3fb3452c521e4ce26e9287d806bff0fda6/Parse/src/test/java/com/parse/ParseInstallationTest.java#L159)) because they don't create the right conditions for the fetch to work (a CACHE_MISS exception is then thrown). Maybe we should update those tests instead, by creating the right conditions, and let the exception propagate by using onSuccessTask at line 1451 as on my [previous commit](https://github.com/parse-community/Parse-SDK-Android/pull/831/commits/605e8a25933c4352f0cd8ef2190f63dd1447308c#diff-8e759e84d8f2c4f7fa6bc715e3786808R1451).

Reply:
Yep, I think it is best to update the tests to make them have the right conditions for success/expected failure rather than modifying the library to work with the expectations of the tests. I think we can go ahead and merge this in though, as it fixes the issue at hand, and keep this in mind for potentially another pull request if that is cool with you.